typescript/src/util.ts

```diff
@@ -221,144 +221,35 @@ export function getResourceNameByPattern(pattern: string): string {
   return name.join('_');
 }
 
-export function isStar(pattern: string): boolean {
-  return pattern === '*';
-}
-
-export function isDoubleStar(pattern: string): boolean {
-  return pattern === '**';
-}
-
-// This intakes an array of strings and checks if there is at least one and only one named segment.
-// Named segments can be resource names or wildcards.
-export function checkIfArrayContainsOnlyOneNamedSegment(
-  pattern: string[]
-): boolean {
-  let countOfNamedSegments = 0;
-  // Checks that resources are named
-  const regexNamed = new RegExp(/^.+=.+/);
-  const regexWildcard = new RegExp(/^{[a-zA-Z\d-]+}/);
-  pattern.forEach(element => {
-    if (regexNamed.test(element) || regexWildcard.test(element)) {
-      countOfNamedSegments++;
-    }
+export function processPathTemplate(
```

Comment:
Would you mind adding a comment for this function? I'm not quite sure what it does, or how it replaces all the deleted functions in TS.

Reply:
Sure!